File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.cpp
class Solution {
public:
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int result = -1, min_distance = INT_MAX;
        for (int i = 0; i < points.size(); i++) {
            int a = points[i][0], b = points[i][1];
            if (x == a || y == b) {
                int distance = abs(x - a) + abs(y - b);
                if (distance < min_distance) {
                    min_distance = distance;
                    result = i;
                }
            }
        }
        return result;
    }
};

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        min_distance= float('inf')
        index=-1
        for i in range (len(points)):
            if points[i][0]==x or points[i][1]==y:
                distance= abs(points[i][0]-x) + abs(points[i][1]-y)
                if distance<min_distance:
                    min_distance = distance
                    index=i
        return index